packages/flutter/lib/src/cupertino/text_field.dart

```diff
@@ -905,7 +905,7 @@ class _CupertinoTextFieldState extends State<CupertinoTextField> with Restoratio
     final Color? decorationColor = CupertinoDynamicColor.resolve(widget.decoration?.color, context);
 
     final BoxBorder? border = widget.decoration?.border;
-    Border resolvedBorder = border as Border;
+    Border? resolvedBorder = border as Border?;
```

Comment:
the null check in line 915 below shouldn't be necessary anymore, right?

Reply:
Can you elaborate? I don't see check on resolvedBorder at L915